test/session_tests/DocumentStoreTest.ts

```diff
@@ -182,6 +182,16 @@ describe('Document store test', () => {
       expect(product['@metadata']['Raven-Node-Type']).to.not.exist;  
       expect(product['@metadata']['@collection']).to.equals('@empty');      
     });
+
+    it('shouldn\'t set id on plain object without id attribute defined', async () => {
+      let product: Product = <Product>{name: "New Product"};    
+
+      session = store.openSession({requestExecutor});      
+      await session.store<Product>(product);
+      await session.saveChanges();
+
+      expect(product.id).to.be.empty;      
```

Comment:
@a-serdukov-mobilunity @gregolsky I've talked with @ayende and if possible we would like to have the same behavior in all clients regarding sending the 'Identity property'. So when entity is saved (saveChanges) we want to removed the 'Identity property' from the JSON send, when it is retrieved via session then we want to fill the 'Identity property' from @metadata[@id]. Is it doable?